util/resolver/resolver.go

```diff
@@ -38,6 +45,31 @@ func NewResolveOptionsFunc(m map[string]RegistryConf) ResolveOptionsFunc {
 			}
 		}
 
+		if c.ExtraCA != "" {
+			// if a CA is specified for this registry, add it to the trusted CA pool.
+			rootCAs, _ := x509.SystemCertPool()
+			if rootCAs == nil {
+				rootCAs = x509.NewCertPool()
+			}
+			certs, err := ioutil.ReadFile(c.ExtraCA)
+			if err != nil {
+				log.Fatalf("Failed to append %q to RootCAs: %v", c.ExtraCA, err)
```

Comment:
Use `logrus.Errorf` && `logrus.Warnf` (line 59) here as these are already initialized based on the `buildkitd` flags.